clang/test/Sema/attr-nonblocking-sema.cpp

```diff
@@ -0,0 +1,126 @@
+// RUN: %clang_cc1 -fsyntax-only -fblocks -fcxx-exceptions -verify %s
+// RUN: %clang_cc1 -fsyntax-only -fblocks -verify -x c -std=c23 %s
```

Comment:
Some pre-C23 tests would also be nice (possibly in a separate file?); especially for prototypeless functions.

Reply:
There's now a separate file for pre-C23 - currently it just tests the prototypeless case.

I'm having trouble thinking of more things to test for pre-C23 that are different from what's covered in this file for C23.

Reply:
Prototypeless functions are the only thing I can think of as well candidly.